sql-plugin/pom.xml

```diff
@@ -102,6 +102,12 @@
             <artifactId>mockito-core</artifactId>
             <scope>test</scope>
         </dependency>
+
+        <dependency>
+            <groupId>com.nvidia</groupId>
+            <artifactId>rapids-4-spark-hybrid_${scala.binary.version}</artifactId>
```

Comment:
Dependency hierarchy:  Spark-Rapids -> Hybrid(C2C) jar (provided scope)  -> Gluten jar (provided scope)
Hybrid does not pull in gluten/velox. Hybrid jar size is less than 1M.
For external customer building of Spark-Rapids with Hybrid(C2C)  feature, the building will not download Gluten jar. Because Maven does not download indirect transitive dependencies of a provided dependency during the build process.

> Like I said previously I don't think that the plan is to have this jar on the classpath all of the time.

Will change to `provided` scope. 

> We need to have a way to make sure that we can run without velox on the classpath. 

Yes. Will add checking for existence of the jar. If there is no Hybrid jar, will not report Class Not Found error.

> Does this jar pull in gluten/velox? Does it do it dynamically?

This jar does not pull in gluten/velox.  Will add checking for the existence of the Gluten jar.

Reply:
Done: change to provided scope

Reply:
We do not plan to add a MVN profile to disable the Hybrid building. Is this OK?
Instead, we will do the following:
This impl will print clear for the following cases.
From Gary:

- Verify there's no problem to disable velox scan by config and run Scan without providing the hybrid jar. It's to verify the correctness of reflection used in velox scan in plugin code. Any dependent classes in hybrid jar shouldn't be imported when disabling velox scan config.
- Add code to disable the velox scan in unsupported Spark versions and OS whatever the velox config is enabled or disabled. Stop running if user enables the velox scan on unsupported Spark or OS.
- When enabling velox scan on supported OS and Spark, add code to check if the necessary jars are provided correctly.(hybrid should be provided, gluten and 3rdparty jar should be built from correctly Spark version and OS.) Stop running if the jars are missing or wrong.
- Scala 2.12 and 2.13 support (If we don't support, we should add code to check as well. If we do support, we should prebuild the gluten jar as well.)
- JDK 8 and 17 support (If we don't support, we should add code to check as well. If we do support, we should prebuild the gluten jar as well.)

Reply:
Sounds like a good plan. The main thing is that we have a way to run without the velox jars on the classpath.